modules/nested-tabs/widgets/nested-tabs.php

```diff
@@ -1089,39 +1089,62 @@ protected function content_template() {
 			var elementUid = view.getIDInt().toString().substr( 0, 3 ); #>
 			<div class="e-n-tabs-heading" role="tablist">
 				<# _.each( settings['tabs'], function( item, index ) {
-				var tabCount = index + 1,
-				tabUid = elementUid + tabCount,
-				tabTitleKey = 'tab-title-' + tabUid;
-				tabIcon = elementor.helpers.renderIcon( view, item.tab_icon, { 'aria-hidden': true }, 'i' , 'object' );
+				let tabCount = index + 1,
+					tabUid = elementUid + tabCount,
+					tabWrapperKey = tabUid,
+					tabTitleKey = 'tab-title-' + tabUid,
+					tabIconKey = 'tab-icon-' + tabUid,
+					tabActiveIconKey = 'tab-active-icon-' + tabUid,
+					tabIcon = elementor.helpers.renderIcon( view, item.tab_icon, { 'aria-hidden': true }, 'i' , 'object' ),
+					tabActiveIcon = tabIcon,
+					tabId = 'e-n-tab-title-' + tabUid;
 
-				let tabActiveIcon = tabIcon;
 				if ( '' !== item.tab_icon_active.value ) {
 					tabActiveIcon = elementor.helpers.renderIcon( view, item.tab_icon_active, { 'aria-hidden': true }, 'i' , 'object' );
 				}
 
-				let tabId = 'e-n-tab-title-' + tabUid;
 				if ( '' !== item.element_id ) {
 					tabId = item.element_id;
 				}
 
+				view.addRenderAttribute( tabWrapperKey, {
+					'id': tabId,
+					'class': [ 'e-n-tab-title','e-normal' ],
+					'data-tab': tabCount,
+					'role': 'tab',
+					'tabindex': 1 === tabCount ? '0' : '-1',
+					'aria-controls': 'e-n-tab-content-' + tabUid,
+					'aria-expanded': 'false',
+				} );
+
 				view.addRenderAttribute( tabTitleKey, {
-				'id': tabId,
-				'class': [ 'e-n-tab-title','e-normal' ],
-				'data-tab': tabCount,
-				'role': 'tab',
-				'tabindex': 1 === tabCount ? '0' : '-1',
-				'aria-controls': 'e-n-tab-content-' + tabUid,
-				'aria-expanded': 'false',
-				'data-binding-type': 'repeater-item',
-				'data-binding-repeater-name': 'tabs',
-				'data-binding-setting': 'tab_title',
-				'data-binding-index': tabCount,
+					'class': [ 'e-n-tab-title-text' ],
+					'data-binding-type': 'repeater-item',
+					'data-binding-repeater-name': 'tabs',
+					'data-binding-setting': [ 'tab_title' ],
+					'data-binding-index': tabCount,
+				} );
+
+				view.addRenderAttribute( tabIconKey, {
+					'class': [ 'e-n-tab-icon' ],
+					'data-binding-type': 'repeater-item',
+					'data-binding-repeater-name': 'tabs',
+					'data-binding-setting': [ 'tab_icon.value' ],
+					'data-binding-index': tabCount,
+				} );
+
+				view.addRenderAttribute( tabActiveIconKey, {
+					'class': [ 'e-n-tab-icon', 'e-active' ],
+					'data-binding-type': 'repeater-item',
+					'data-binding-repeater-name': 'tabs',
```

Comment:
I will begin by saying that this is the first time I see this templating code. 
I played with it a bit,  Do we need `class`, `data-binding-type` and `data-binding-repeater-name`?
What is their role in the data updating process?
I deleted them and it looks like they don't affect the functionality.
We can chat about that.

Same for all the `addRenderAttribute... ` above

Reply:
@davseve 

`Class`: generates the class for the item. We need it.

`data-binding`-arguments: create the data-binding effect. We need them to create the data-binding effect.

`type` and `name`: I assume that we need them. The `type` is referenced inside `wp-content\plugins\elementor\assets\dev\js\editor\elements\views\base.js`

`data-binding-repeater-name`: I am not sure if this argument is required.

Reply:
@davseve I have have tested it. It doesn't work well when we leave out some of the binding attributes.